lib/orogen/gen/typekit.rb

```diff
@@ -1994,7 +2006,13 @@ def generate
 
                 # Generate opaque-related stuff first, so that we see them in
                 # the rest of the typelib-registry-manipulation code
+                #
+                # First make sure we've loaded everything
+                perform_pending_loads
                 handle_opaques_generation(registry)
+                # Make sure that any pending load added by
+                # handle_opaques_generation is actually loaded. We need a
+                # complete registry after this point
                 perform_pending_loads
```

Comment:
Are there guarantees that after the last call to perform_pending_loads no further opaque handling is needed? 
Otherwise loop until no pending_loads are left  and no opaque handling is required?

Reply:
Good to merge then.